### ISSUE TITLE
Input settings are not available on iOS

### ISSUE THREAD

**Issue by Susko3** [ppy/osu#25368]
### Type

Other

### Bug description

Noticed when working on https://github.com/ppy/osu/pull/25366.

Since input settings are hardcoded to only appear on desktop, they won't show on iOS. Even though iOS uses the same SDL2 input handlers as desktop.

https://github.com/ppy/osu/blob/df08c4e1adf701068e9c00dd2e2fd40d248c9bb2/osu.Game/OsuGameBase.cs#L566-L582

I don't have an iOS dev environment so I can't test and judge whether these settings actually make sense on iOS, so I've left the exact implementation to someone who can test.

I've highlighted in green settings that I think are useful on iOS as well:

![image](https://github.com/ppy/osu/assets/16479013/e991b7a0-29b8-4d99-89bd-b4bf2ec88195)

### Version

2023.1026.0

### Logs

-

**Comment by steve-xiao** [comment]
Seconded. Kinda frustrated about this being a hover and click player

**Comment by Susko3** [comment]
@steve-xiao what input method (pen, mouse, touch, etc.) are you using, and what settings are you missing?

**Comment by steve-xiao** [comment]
I'm missing all the functions you enclosed in green boxes. I've only been able to use touch input on ios

**Comment by steve-xiao** [comment]
To be more precise, I'm missing all functions on your screenshot except MIDI since touch input is enforced

**Comment by Susko3** [comment]
> I'm missing all the functions you enclosed in green boxes. I've only been able to use touch input on ios

Do you mean that mouse or keyboard input is not working?
This is a bug report about settings, the mouse handler should be working just fine, only that you can't configure it.

> since touch input is enforced

Could you elaborate what you mean by that.
osu! should support MIDI, controllers, keyboard, touch and mouse on iOS (tablet drivers shouldn't work).

**Comment by steve-xiao** [comment]
Sorry for the confusion, just tested and found out mouse input does work lol. It's just that I can't get it to read hovering

**Comment by Susko3** [comment]
> Sorry for the confusion, just tested and found out mouse input does work lol. It's just that I can't get it to read hovering

Could you record a video of hover not working?

Is the below an accurate description of what you're seeing?
- clicking with a mouse will move the mouse cursor and register the click as expected
- just moving the mouse will not update the cursor position

**Comment by steve-xiao** [comment]
No, as I said, mouse input does work both for clicking and updating the cursor position. I mean I can't get it to register hovering from a pen as the source of cursor position, like how the PC version of osu would register hovering of a drawing tablet pen. I can get it to register dragging with a pen just fine though. I'll post a video later about this.

**Comment by steve-xiao** [comment]
Just to clear any confusion, when I said I was lacking all the functions you highlighted, I just meant that the settings for them are not visible on ios, and I then assumed that not being able to configure the mouse input on ios would mean that a mouse would not work on it, which was wrong.

**Comment by peppy** [comment]
This was intentional.

None of these settings make sense for now, except maybe joystick. Mouse button toggle is a legacy one and can be achieved via bindings.

Mouse handling on iOS is limited by the API implementation, there's no acceleration we can set currently.

Dunno, tempted to leave this in discussion state until someone actually needs something.

**Comment by OiviaDesu** [comment]
I conducted a brief experiment with some input devices such as pen, keyboard and mouse on iPhone 15 using a USB-C adapter. The results were mixed. The keyboard allowed me to type and use key bindings, but it did not function in the gameplay. The mouse worked normally, but I could not adjust the sensitivity. The pen also worked, but it only registered taps and not presses, and I could not modify the range either. Here are the devices I used:

- Logitech Keyboard G PRO X
- Logitech Mouse G502 Hero
- XP-PEN G430s
